npm/src/upload/upload-form.ts

```diff
@@ -25,6 +27,36 @@ export class UploadForm {
     return value
   }
 
+  addFolderListener() {
+    this.folderRetriever.addEventListener("change", () => {
+      const form: HTMLFormElement | null = this.formElement
```

Comment:
I forgot to add my findings. @suzannehamilton, based on my research, it doesn't seem as though you can't mutate a `FileList`. And you can't replace it with a mock one as it doesn't have the necessary methods a `FileList` has. Not sure if maybe we want to try creating an interface that extends the `FileList` and then if that works, add the files to that object?

If that doesn't work, think the way I had it before with the `this.formElement` is the only way it will work on the frontend and in the tests because in the tests, I can easily assign `uploadForm.files` (`uploadForm` is the parameter that is passed in and becomes `formElement`) to a fake object because it's as if the tests don't know that `uploadForm.files` (AKA `this.formElement.files`) and `folderRetriever` are the same thing. I'm not sure why....they must be the same thing because when I run the frontend, it can get the files out.